lib/runtime/wazero/imports.go

```diff
@@ -817,49 +820,48 @@ func ext_trie_blake2_256_root_version_1(ctx context.Context, m api.Module, dataS
 		return 0
 	}
 
-	hash, err := t.Hash()
-	if err != nil {
-		logger.Errorf("failed computing trie Merkle root hash: %s", err)
-		return 0
-	}
-
 	logger.Debugf("root hash is %s", hash)
 	m.Memory().Write(ptr, hash[:])
 	return ptr
 }
 
 func ext_trie_blake2_256_ordered_root_version_1(ctx context.Context, m api.Module, dataSpan uint64) uint32 {
+	return ext_trie_blake2_256_ordered_root_version_2(ctx, m, dataSpan, 0)
+}
+
+func ext_trie_blake2_256_ordered_root_version_2(
+	ctx context.Context, m api.Module, dataSpan uint64, version uint32) uint32 {
 	rtCtx := ctx.Value(runtimeContextKey).(*runtime.Context)
 	if rtCtx == nil {
 		panic("nil runtime context")
 	}
 
 	data := read(m, dataSpan)
 
-	t := trie.NewEmptyTrie()
+	stateVersionBytes, _ := m.Memory().Read(version, 4)
+	stateVersion, err := trie.ParseVersion(binary.LittleEndian.Uint32(stateVersionBytes))
+	if err != nil {
+		logger.Errorf("failed parsing state version: %s", err)
+		return 0
+	}
+
```

Comment:
You don't need to read the version from memory, the state version is already given to you through the `version` argument, the version is either `0` or `1`.

https://spec.polkadot.network/chap-host-api#id-version-2-prototype-9

So you can just:

```suggestion
	stateVersion, err := trie.ParseVersion(version)
	if err != nil {
		logger.Errorf("failed parsing state version: %s", err)
		return 0
	}
```

Reply:
Very good point, since this is my first time changing the runtime I have some questions regarding the way we pass the parameters.

I'm doing this in the test:

```go
stateVersion := uint32(trie.V1)
stateVersionBytes := make([]byte, 4)
binary.LittleEndian.PutUint32(stateVersionBytes, stateVersion)
```

What is the right way to pass it as integer instead of bytes? or do I have to change anything else?

Reply:
I maybe need to change the [changes](https://github.com/ChainSafe/polkadot-spec/blob/c6fa5767265e022f0fc5cd0c50a8c3e97b8a65bb/test/runtimes/hostapi/src/lib.rs#L621C41-L621C41) I made in our test runtime too?

Reply:
I think we don't need to change the testing, when you call a runtime function you need to parse everything to bytes.